### PR TITLE
LPS-199503 Support export of Object entries at site level

### PR DESCRIPTION
**Pull request by matijapetanjek**

### CODE REVIEW

**Comment by matijapetanjek** [comment]
ci:test:relevant

**Review by sergiojimcos — APPROVED** [review]
LGTM!!

File: modules/apps/batch-planner/batch-planner-rest-impl/src/main/java/com/liferay/batch/planner/rest/internal/resource/v1_0/SiteScopeResourceImpl.java [hunk 1]
```diff
@@ -41,6 +44,15 @@ public Page<SiteScope> getPlanInternalClassNameKeySiteScopesPage(
 			String internalClassNameKey, Boolean export)
 		throws Exception {
 
+		int index = internalClassNameKey.indexOf(StringPool.POUND);
+
+		if (index > 0) {
+			return Page.of(
+				_getSiteScopes(
+					_getObjectScope(
+						internalClassNameKey.substring(index + 1))));
```
[the remark this answers is not in the document]
Comment (ccorreagg) [reply modules/apps/batch-planner/batch-planner-rest-impl/src/main/java/com/liferay/batch/planner/rest/internal/resource/v1_0/SiteScopeResourceImpl.java:53]: Sorry, I meant:

```
TaskItemUtil.getTaskItemDelegateName(internalClassNameKey)
```

File: modules/apps/batch-engine/batch-engine-api/src/main/java/com/liferay/batch/engine/jaxrs/uri/BatchEngineUriInfo.java [hunk 2]
```diff
@@ -150,7 +150,13 @@ public Builder taskItemDelegateName(String taskItemDelegateName) {
 
 	private BatchEngineUriInfo(Builder builder) {
 		_pathParameters = builder._pathParameters;
+
 		_queryParameters = builder._queryParameters;
+
+		if (_queryParameters.containsKey("siteId")) {
```
Comment (ccorreagg) [inline modules/apps/batch-engine/batch-engine-api/src/main/java/com/liferay/batch/engine/jaxrs/uri/BatchEngineUriInfo.java:156]: I am not sure about including this if clause here, as we are considering the `scopeKey` parameter is going to be always set with the `siteId` value.

What do you think about considering always in the Batch layer `siteId` as `siteId`, and then updating the implementation of the `read` method of the VulcanBatch interface to change the way the `scopeKey` value is read from the parameters in the `ObjectEntryResourceImpl`, considering both options?.

Something like:

```
@Override
public Page<ObjectEntry> read(
		Filter filter, Pagination pagination, Sort[] sorts,
		Map<String, Serializable> parameters, String search)
	throws Exception {

	_loadObjectDefinition(parameters);

	ObjectScopeProvider objectScopeProvider =
		_objectScopeProviderRegistry.getObjectScopeProvider(
			_objectDefinition.getScope());

	if (objectScopeProvider.isGroupAware()) {
		return getScopeScopeKeyPage(
			_getScopeKey(parameters),
			Boolean.parseBoolean((String)parameters.get("flatten")), search,
			null, filter, pagination, sorts);
	}

	[...]

}
```
being:
```
private String _getScopeKey(Map<String, Serializable> parameters) {
	if (parameters.containsKey("scopeKey")) {
		return String.valueOf(parameters.get("scopeKey"));
	}

	if (parameters.containsKey("siteId")) {
		return String.valueOf(parameters.get("siteId"));
	}

	return null;
}

```

File: modules/apps/batch-planner/batch-planner-test/src/testIntegration/java/com/liferay/batch/planner/batch/engine/broker/test/BatchEngineBrokerTest.java [hunk 3]
```diff
@@ -370,11 +373,79 @@ public void testExportObjectDefinition() throws Exception {
 			actualJsonNode);
 	}
 
+	@Test
```
Comment (ccorreagg) [inline modules/apps/batch-planner/batch-planner-test/src/testIntegration/java/com/liferay/batch/planner/batch/engine/broker/test/BatchEngineBrokerTest.java:376]: Maybe it is the time (not sure if this PR or a following-up PR) to include tests for the `SiteScopeResourceImpl`

**Comment by ccorreagg** [comment]
See https://github.com/liferay-headless/liferay-portal/pull/1618

**Pull request closed without merging**